src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -40,6 +42,8 @@ public Http2Stream(Http2Connection connection)
                 _disposed = false;
 
                 _responseBuffer = new ArrayBuffer(InitialBufferSize);
+
+                _streamWindow = new CreditManager(InitialWindowSize);
```

Comment:
It might be worthwhile to handle SETTINGS_INITIAL_WINDOW_SIZE updates for better efficiency. 

By default, Kestrel uses an initial stream window size of 96KB instead of the 65535 protocol default. And while Kestrel doesn't currently support this, the protocol allows the initial window size to be made smaller, so this could potentially help avoid a protocol violation as well.

Reply:
Yeah, there's an issue to implement this. #34059.